torch/utils/_sympy/functions.py

```diff
@@ -589,8 +617,53 @@ def __new__(cls, *args, **assumptions):
         # base creation
         obj = Expr.__new__(cls, *ordered(args), **assumptions)
         obj._argset = args
+
+        if unique_summation_atoms:
+            obj.unique_summation_atoms = unique_summation_atoms
         return obj
 
+    @classmethod
+    def _should_optimize_max_of_unique_summations(cls, args):
```

Comment:
Please annotate with input/return type (really, the return type is the important one here)

Reply:
done